### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by chimwemwe007**
### Created pages
- Create a rocket page and fetch the rocket date from API to dispatch an action to store the selected data in the Redux store:
- Created mission page and fetch mission data from API to dispatch an action to store the selected data in Redux store:
- Created profile page and fetch profile data from API and display selected joined and leave the mission. reserve rocket 

### Used Redux to do the following:
- Used seSelector(), Redux Hook, to select the state slices and render lists of rockets and missions in corresponding routes.
- Rendered a table with the missions' data (as per design).
- Created actions and reducers for booking rockets and joining missions.
- Created actions and reducers for canceling rockets and leaving missions.

### CODE REVIEW

File: src/components/ProfileComponents.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { useSelector } from 'react-redux';
+
+function ProfileComponent() {
+  const states = useSelector((state) => state);
+  let rocketsReserved = 0;
+  let missionJoined = 0;
+  return (
+    <div className="container">
+      <br />
+      <div className="row">
```
Comment (BenMukebo) [inline src/components/ProfileComponents.js:10]: - [OPTIONAL] The My Profile section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections. 👍🏻 

- [OPTIONAL] The My Profile section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload. 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi @Lordkaito, @hamayounali5, @Hamayounali,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉

- Git flow is followed. :heavy_check_mark:
- You have followed  JavaScript best practices. ✔️ 
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- `useSelector()` was used to get a state slice corresponding to routes. ✔️ 
- Renders a list of rockets (as per design). ✔️ 
- Renders a table with the missions' data (as per design). ✔️ 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1 +1,89 @@
...
+
+- GitHub: [@Hamayounali](https://github.com/Hamayounali)
+- Twitter: [@hamayounAli6](https://twitter.com/hamayounAli6)
+- LinkedIn: [Humayoon Ali](https://www.linkedin.com/in/humayoon-ali-663ba2239)
+
+👤 **Chimwemwe Mkandawire**
+
+- GitHub: [@chimwemwe007](https://github.com/chimwemwe007)
+- Twitter: [@CHXMZ](https://twitter.com/chxmz)
+- LinkedIn: [Chimwemwe Mkandawire](https://www.linkedin.com/in/Chimwemwe-Mkandawire)
```
[the remark this answers is not in the document]
Comment (chimwemwe007) [reply README.md:70]: Hello @BenMukebo ,
- There are 2 group member in the team, we both just have 2 accounts linked to us both and we dont know how to deactivate our accounts. It is just 2 people both with an extra account.

File: README.md [hunk 3]
```diff
@@ -1 +1,89 @@
...
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the issues page.
```
[the remark this answers is not in the document]
Comment (chimwemwe007) [reply README.md:79]: Fixed

File: src/components/Navbar.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        <ul className="navbar-nav ms-auto">
+          <li className="nav-item">
+            <Link className="nav-link" to="/">Rockets</Link>
+          </li>
+          <li className="nav-item">
+            <Link className="nav-link" to="/missions">Missions</Link>
+          </li>
+          <li className="nav-item">
+            <Link className="nav-link" to="/profile">My Profile</Link>
```
[the remark this answers is not in the document]
Comment (Hamayounali) [reply src/components/Navbar.js]: Thank you @BenMukebo, I have updated the link as you guided.

File: src/components/Navbar.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+        </Link>
+
+        <ul className="navbar-nav ms-auto">
+          <li className="nav-item">
+            <Link className="nav-link" to="/">Rockets</Link>
+          </li>
+          <li className="nav-item">
+            <Link className="nav-link" to="/missions">Missions</Link>
+          </li>
```
[the remark this answers is not in the document]
Comment (Hamayounali) [reply src/components/Navbar.js]: Dear @BenMukebo, our team has 2 members, @Hamayounali and @chimwemwe007. Mistakenly we used other GitHub accounts.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You did well to make the changes requested by the previous reviewer✔️✔️

## Optional suggestions

- [ ] Let's take note to correct optional changes as well.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Navbar.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenMukebo) [reply src/components/Navbar.js]: Hey @Hamayounali, Let's ignore the Dragons part then 👍🏻

**Pull request merged**